### PR TITLE
Removing unused CMake checks for version less than 3.5 (#421)

### PR DESCRIPTION
**Pull request by MiladShafiee**
There was some unused checking for the minimum version required of CMake in CMakes build scripts, so we removed these additional checking as described in this issue  #421!

### CODE REVIEW

**Comment by MiladShafiee** [comment]
> Great, thanks a lot.
> 
> As you removed the workaround for the linking of `gazebo_yarp_lib_common`, I think a further step in the simplification would be to remove the set of the `LIB_COMMON_NAME` and directly pass the `gazebo_yarp_lib_common` to the `LINKED_LIBRARIES` option of the `add_gazebo_yarp_plugin_target` CMake function. However, we can also merge as it is, as it is already an improvement.

@traversaro  Thank you! I've done the proposed modification!

File: CMakeLists.txt.user [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+ </data>
+ <data>
+  <variable>ProjectExplorer.Project.Updater.FileVersion</variable>
+  <value type="int">20</value>
+ </data>
+ <data>
+  <variable>Version</variable>
+  <value type="int">20</value>
+ </data>
+</qtcreator>
```
Comment (traversaro) [inline CMakeLists.txt.user]: The `CMakeLists.txt.user` file is a file automatically generated by QtCreator, and it is not supposed to be committed to the repo.

Comment (MiladShafiee) [reply CMakeLists.txt.user]: Thank you. I removed this file.

**Comment by traversaro** [comment]
Thanks!

**Pull request merged**